### ISSUE TITLE
sort_by_fields() fails when model is ordered through a foreign key

### ISSUE THREAD

**Issue by juyrjola** [wagtail/django-modelcluster#45]
If the `ordering` field of a model is configured to order the objects over a foreign key relationship (using the `__` syntax), sort_by_fields() fails.

E.g.

``` python
   class Car(models.Model):
   ...
   class Meta:
       ordering = ['brand__name']
```

**Comment by elbaschid** [comment]
I am running into the same issue with using a `ParentalKey` in the `ordering` setting:
```
class Vineyard(Model):
    name = CharField(max_length=200)


class Dinner(ClusterableModel):
    vineyard = ParentalKey(Vineyard)
    class Meta:
        ordering = ('vineyard',)
```

This raises an exception with the following message:
```
  File "$HOME/lib/python3.5/site-packages/modelcluster/utils.py", line 19, in sort_by_fields
    items.sort(key=lambda x: (getattr(x, key) is not None, getattr(x, key)), reverse=reverse)
TypeError: unorderable types: Vineyard() < Vineyard()
```

One way to make this work is to implement `__gt__` and `__lt__` on the `Vineyard` model to make them comparable. Something like this works for me:
```
class Vineyard(Model):
    name = CharField(max_length=200)

    def __lt__(self, other):
        return self.name < other.name

    def __gt__(self, other):
        return self.name > other.name
```

I'm not quite sure if that would be the suggested solution or more of a workaround. I would think an implementation that works with Django model out of the box would be best but implicitly comparing model without adding additional DB queries might not be straight forward.

Are there any alternative suggestions?

**Comment by thenewguy** [comment]
I've encountered this as well attempting to order based on a field of the foreign key relation as well.  My use case is attempting to use the InlinePanel on a "through" model to mimic a m2m relation.  My intentions are to order based on a tag:
```
class TagFilter(models.Model):
    class Meta:
        ordering = ['tag__slug']
    
    tag = models.ForeignKey(Tag, related_name="+")
    page = ParentalKey(TagFilterPage, related_name='tag_filters')

```

**Comment by thenewguy** [comment]
Btw, my workaround for the meantime is just to duplicate the data.  Like this:
```
class Meta:
        ordering = ['_duplicate_tag_slug']

_duplicate_tag_slug = models.SlugField(max_length=100, editable=False)
```
And then set the data when tags are saved or the tagfilter is saved.  Works easily enough with signals for simple ordering.

**Comment by przemub** [comment]
The problem is still there. My workaround without duplicating the data is:
```python
class Category(models.Model):
    name = models.CharField(unique=True, blank=False, max_length=50)
    type = models.ForeignKey(
        CategoryType, null=False, on_delete=models.CASCADE
    )

    @property
    def type__name(self):
        """
        Circumvents https://github.com/wagtail/django-modelcluster/issues/45
        """
        return self.type.name

    class Meta:
        ordering = ["type__name", "name"]
```

**Comment by gasman** [comment]
This has now been implemented in #148 (but isn't in a release yet, so you'll need to install directly from git).

**Comment by przemub** [comment]
@gasman Awesome, thank you a lot for letting me know!
